src/annotation.js

```diff
@@ -0,0 +1,607 @@
+var $ = require('jquery');
+var inherit = require('./inherit');
+var geo_event = require('./event');
+var transform = require('./transform');
+
+var annotationId = 0;
+
+var annotationState = {
+  create: 'create',
+  done: 'done',
+  edit: 'edit'
+};
+
+/////////////////////////////////////////////////////////////////////////////
+/**
+ * Base annotation class
+ *
+ * @class geo.annotation
+ * @param {string} type the type of annotation.
+ * @param {object?} options Inidividual annotations have additional options.
+ * @param {string} [options.name] A name for the annotation.  This defaults to
+ *    the type with a unique ID suffixed to it.
+ * @param {geo.annotationLayer} [options.layer] a reference to the controlling
+ *    layer.  This is used for coordinate transforms.
+ * @param {string} [options.state] initial annotation state.  One of the
+ *    annotation.state values.
+ * @returns {geo.annotation}
+ */
+/////////////////////////////////////////////////////////////////////////////
+var annotation = function (type, args) {
```

Comment:
Right, I saw the code " annotation.call(this, 'rectangle', args);" so the type names are defined by each subclass. I am wondering if could return a list of supported types (assuming after the annotations are initialized) as something like this could be useful for ui. But I don't think this is critical or needed at the moment. May be (no strong preference) in the comment, we could add a line in the doc stating that type is defined by sub-classes as they extend the base annotation class.